codec/testdata/test_service.go

```diff
@@ -0,0 +1,19 @@
+package testdata
```

Comment:
Why is package named `testdata`? It could be very confusing, let alone that it does not seem to meet Golang's conventions on `testdata` directories [1]

[1] https://golang.org/pkg/cmd/go/internal/test/

Reply:
Can you suggest a better name?

Reply:
`testservice`?

Reply:
Well it's not just the service, it's all out test protobuf data types, thus `testdata`

Reply:
`testdatatypes`?

Anyway, I see. The `go` tool ignores `testdata` directories, so as long as this does not represent a problem, it should be fine.

Reply:
Maybe we keep it for now and if it's a problem we can change later? So far it's been working.